lib/msf/base/config.rb

```diff
@@ -50,6 +50,24 @@ def self.get_config_root
     end
   end
 
+  # Determines the git hash for this source tree
+  #
+  # @return [String] the git hash for this source tree
+  def self.get_git_hash
+    msfrelease = File.join(InstallRoot, 'msfrelease.txt')
+    hash = 'unknown'
+    @git_hash ||= begin
+      if File.exist?(msfrelease)
+        hash = File.read(msfrelease)
+      else
+        if File.exist?(File.join(InstallRoot, '.git'))
+          hash = `git rev-parse HEAD`
```

Comment:
I'd rather not run the command at all if git is not in the PATH than try to guess where it is. Shall I implement 'which' or is there a convenient idiomatic way to do this? I didn't see an exception being thrown with the backtick method. For packaged releases, we'll have an msfrelease.txt file anyway, bypassing this.